### ISSUE TITLE
[Enhancement] `newline` argument for `csv.reader()`

### ISSUE THREAD

**Issue by opk12** [pylint-dev/pylint#9422]
### Current problem

The doc for [csv.reader()](https://docs.python.org/3/library/csv.html#csv.reader) warns that **not** passing `newline` [can cause issues](https://docs.python.org/3/library/csv.html#id4).

### Desired solution

Warn if `csv.reader()` is not being passed `newline` explicitly.

### Additional context

I found this in a student homework. Is it in-scope for pylint to warn? Note that refurb did **not** warn.

```python
import csv
result = {}

def f():
    with open("/tmp/file1", 'r') as route:
        for row in csv.reader(route, delimiter="\t"):
            counter = counter + 1
            result[row[0]] = row[1]
        ...
    return counter

f()
```
